### ISSUE TITLE
视频生成有noise

### ISSUE THREAD

**Issue by phyllispeng123** [Lakonik/piFlow#8]
您好！！感谢您的工作，我尝试在视频生成模型上使用GMPolicy进行蒸馏，我的方法如下：
1. 在原有的100NFE的7B视频模型上加入`self.mean，self.logweights，self.logstd`的三个线性层。其中`self.mean，self.logweights`的输入为模型的输出(b, seq_len, channel), `self.logstd`输入为time_embedding，其输出分别为(b, self.num_guassians, seq_len, channel), (b, self.num_guassians, seq_len,1)，(b, self.num_guassians, 1, 1) （没有patchfy）
2. 训练：参考piFlow/lakonlab/models/diffusions/piflow.py，配置为`num_guassians = 32, shift=5，lr = 1e-5，nfe = 4，num_decay_iters= 5000， window_substeps=3，gm_dropout = 0.1，num_intermediate_states = 4，final_step_size_scale = 0.5， total_substeps = 128`，其余配置与代码中`PiFlowImitationBase.piid_segment`训练方式 & 代码一致。在训练中我freeze了原始模型，只训练了3个linear 层，但是我没有训练lora层，目的是先排查视频的noise问题
4. 推理：使用的是`PiFlowImitationBase.forward_test`，并应用训练时的参数

结果我生成的视频带有noise，想请教一下我的训练 & 推理流程是否正确？你们是否在训练中遇到相似问题？有没有什么方法可以debug一下？我的loss是all_tgt_u和all_pred_u之间的mse loss，在num_intermediate_states这个维度求的均值（rescale_mode = constant)，发现loss没有收敛

<img width="2539" height="1638" alt="Image" src="https://github.com/user-attachments/assets/bed9a8bc-d748-4ba3-8cea-b2153c890a33" />

**Comment by Lakonik** [comment]
Closing this for now since distilling video models is not yet officially supported in this codebase. Feel free to add more comments for further discussions.

**Comment by YunkangTao** [comment]
> > [@Lakonik](https://github.com/Lakonik) [@phyllispeng123](https://github.com/phyllispeng123) Hi, I try the toymodel to overfitting a trajetory between one noise and one data in wan2.1-1.3b. I find that the generated video is noise, although the loss continue to decrease. Do you have any advice or suggestion about that, please？Or do you have any advances in video model training? Thank you!
> 
> similar to me, i got noise video too

I have resolved this problem by using UniPCMultistepScheduler in diffusers library ranther than ContinuousTimeStepSampler when overfitting one noise to one video on wan-2.1 1.3b model. Maybe you can fix it referring to wan2.1 official inference scripts. Other settings are same as qwen-image piflow toymodel, such as lr, num_gaussians, num_intermediates, etc. I am currently trying to scale the algorithm to wan-2.1 training using lora referring to qwen-image piflow distillation. But I don't know whether I can make it using the UniPCMultistepScheduler. Well, maybe you can try the toymodel overfitting firstly. Keep contacting.

**Comment by YunkangTao** [comment]
> > > > [@Lakonik](https://github.com/Lakonik) [@phyllispeng123](https://github.com/phyllispeng123) Hi, I try the toymodel to overfitting a trajetory between one noise and one data in wan2.1-1.3b. I find that the generated video is noise, although the loss continue to decrease. Do you have any advice or suggestion about that, please？Or do you have any advances in video model training? Thank you!
> > > 
> > > 
> > > similar to me, i got noise video too
> > 
> > 
> > I have resolved this problem by using UniPCMultistepScheduler in diffusers library ranther than ContinuousTimeStepSampler when overfitting one noise to one video on wan-2.1 1.3b model. Maybe you can fix it referring to wan2.1 official inference scripts. Other settings are same as qwen-image piflow toymodel, such as lr, num_gaussians, num_intermediates, etc. I am currently trying to scale the algorithm to wan-2.1 training using lora referring to qwen-image piflow distillation. But I don't know whether I can make it using the UniPCMultistepScheduler. Well, maybe you can try the toymodel overfitting firstly. Keep contacting.
> 
> Thank you for your information, did you try gm_policy or dxpolicy ? I tried the gm_policy and got noise video at step 199, maybe train longer will work? how many steps you train ?

I try gm policy and get some semantic structures at iter500 and a clear video at iter2000.
